microsoft/testsuites/core/storage.py

```diff
@@ -0,0 +1,203 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import re
+from pathlib import PurePosixPath
+from typing import Optional
+
+from assertpy.assertpy import assert_that
+
+from lisa import (
+    LisaException,
+    Logger,
+    RemoteNode,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+    simple_requirement,
+)
+from lisa.environment import EnvironmentStatus
+from lisa.operating_system import CoreOs
+from lisa.tools import Cat, Mount
+from lisa.tools.lsblk import Lsblk
+from lisa.tools.swapon import SwapOn
+
+
+@TestSuiteMetadata(
+    area="storage",
+    category="functional",
+    description="""
+    This test suite is used to run storage related tests.
+    """,
+)
+class Storage(TestSuite):
+    DEFAULT_DISK_TIMEOUT = 300
+
+    # ResourceDisk.MountPoint=/mnt
+    _resource_disk_mount_point_regex = re.compile(
+        r"^.*ResourceDisk\.MountPoint=(?P<resource_disk_mount_point>\S+)\s.*$"
+    )
+
+    # ResourceDisk.EnableSwap=n
+    # ResourceDisk.EnableSwap=y
+    _resource_disk_enable_swap_regex = re.compile(
+        r"^.*ResourceDisk\.EnableSwap=(?P<resource_disk_enable_swap>\S+)\s.*$"
+    )
+
+    # /dev/sda1 on / type ext4 (rw,relatime,discard)
+    _root_partition_regex = re.compile(
+        r"\s*\/dev\/(?P<partition>\D+).*\s+on\s+\/\s+type.*"
+    )
+
+    @TestCaseMetadata(
+        description="""
+        This test will check that VM root disk(os disk) is provisioned
+        with the correct timeout.
+        Steps:
+        1. Find the root disk (os disk) partition for the VM. The root partition
+        corresponds to entry with mount point `/' in the `mount` command.
+        2. Verify the timeout value for root disk in
+        `/sys/block/<partition>/device/timeout` file is set to 300.
+        """,
+        priority=1,
+        requirement=simple_requirement(
+            environment_status=EnvironmentStatus.Deployed,
```

Comment:
The `environment_status` is optional for most test cases, unless a test case like smoke_test, it needs to start before the vm is connected.